packages/augur-core/source/contracts/utility/RepSymbol.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.5.10;
+
+import 'ROOT/IAugur.sol';
+import 'ROOT/IAugurMarketDataGetter.sol';
+import 'ROOT/reporting/IUniverse.sol';
+import 'ROOT/libraries/math/UintToString.sol';
+import 'ROOT/libraries/BytesToString.sol';
+import 'ROOT/utility/IRepSymbol.sol';
+
+
+/**
+ * @title Rep Symbol
+ * @notice A Utility contract used to retrieve the symbol for a REP token. We extract this out to reduce REP token contract size.
+ */
+contract RepSymbol is IRepSymbol {
+    using UintToString for uint;
+    using BytesToString for bytes32;
+
+    function getRepSymbol(address _augurAddress, address _universeAddress) external view returns (string memory) {
+        IAugur _augur = IAugur(_augurAddress);
+        IUniverse _universe = IUniverse(_universeAddress);
+        IUniverse _parentUniverse = _universe.getParentUniverse();
+
+        if (_parentUniverse != IUniverse(0)) {
+            uint256 _forkIndex = _augur.getUniverseForkIndex(_parentUniverse);
+            IMarket _forkingMarket = _parentUniverse.getForkingMarket();
+            uint256 _numTicks = _forkingMarket.getNumTicks();
+            uint256[] memory _payoutNumerators = _universe.getPayoutNumerators();
+            if (_payoutNumerators[0] != 0) {
+                return string(abi.encodePacked("REPv2", "_", _payoutNumerators[0] == _numTicks ? "INVALID" : "MALFORMED", "_", _forkIndex.uint2str()));
+            }
+            IMarket.MarketType _marketType = IAugurMarketDataGetter(_augurAddress).getMarketType(_forkingMarket);
+            string memory _outcome = "YES";
+            if (_marketType == IMarket.MarketType.YES_NO) {
+                if (_payoutNumerators[1] == _numTicks) {
+                    _outcome = "NO";
+                } else if (_payoutNumerators[1] != _numTicks) {
+                    _outcome = "MALFORMED";
+                }
+            } else if (_marketType == IMarket.MarketType.CATEGORICAL) {
+                uint256 _numOutcomes = _forkingMarket.getNumberOfOutcomes();
+                bytes32[] memory _outcomes = IAugurMarketDataGetter(_augurAddress).getMarketOutcomes(_forkingMarket);
+                for (uint256 _i = 1; _i < _numOutcomes; _i++) {
+                    if (_payoutNumerators[_i] != _numTicks) {
+                        _outcome = "MALFORMED";
+                    } else if (_payoutNumerators[_i] != 0) {
```

Comment:
Is this conditional necessary, given that _numTicks is always positive anyway?

Reply:
the payout for all non winning outcomes will be 0 so this is to skip them

Reply:
They're all being chosen in the first case though since 0 != _numTicks.

Reply:
Ah, right, actually this shouldn't work at all. Will refactor a bit

Reply:
In the process of refactoring found a different bug here actually! thanks for drawing attention to it